native/src/main/java/io/ballerina/stdlib/log/MaskedStringBuilder.java

```diff
@@ -0,0 +1,578 @@
+    }
+
+    /**
+     * Reset the builder state for reuse while keeping the allocated memory.
+     * This is more efficient than creating a new builder instance.
+     */
+    public void reset() {
+        if (this.closed) {
+            throw ErrorCreator.createError(MASKED_STRING_BUILDER_HAS_BEEN_CLOSED);
+        }
+        this.visitedValues.clear();
+        this.stringBuilder.setLength(0);
+        this.escapeBuffer.setLength(0);
+    }
+
+    /**
+     * Check if the builder has been closed.
+     *
+     * @return true if closed, false otherwise
+     */
+    public boolean isClosed() {
+        return this.closed;
+    }
+
+    @Override
+    public void close() {
+        if (!this.closed) {
+            this.visitedValues.clear();
+            this.stringBuilder = null;
+            this.escapeBuffer = null;
+            this.closed = true;
+        }
+    }
+
+    /**
+     * Clear the annotation cache to free memory.
+     * Should be called periodically in long-running applications.
+     */
+    public static void clearAnnotationCache() {
+        ANNOTATION_CACHE.clear();
+    }
+
+    /**
+     * Get the size of the annotation cache for monitoring purposes.
+     */
+    public static int getAnnotationCacheSize() {
+        return ANNOTATION_CACHE.size();
+    }
+
+    /**
+     * Create a new MaskedStringBuilder instance with default settings.
+     *
+     * @param runtime the Ballerina runtime
+     * @return a new MaskedStringBuilder instance
+     */
+    public static MaskedStringBuilder create(Runtime runtime) {
+        return new MaskedStringBuilder(runtime);
+    }
+
+    /**
+     * Create a new MaskedStringBuilder instance with specified initial capacity.
+     *
+     * @param runtime the Ballerina runtime
+     * @param initialCapacity the initial capacity for the internal StringBuilder
+     * @return a new MaskedStringBuilder instance
+     */
+    public static MaskedStringBuilder create(Runtime runtime, int initialCapacity) {
+        return new MaskedStringBuilder(runtime, initialCapacity);
+    }
+
+    static Optional<BMap<?, ?>> getLogSensitiveDataAnnotation(Map<String, BMap<?, ?>> fieldAnnotations,
+                                                              String fieldName) {
+        // In the value map keys are unescaped, but the annotation keys are escaped
+        // Moreover runtime does not provide a way to unescape the annotation keys, so we need to escape the field name
+        BMap<?, ?> fieldAnnotationMap = fieldAnnotations.get(IdentifierUtils.escapeSpecialCharacters(fieldName));
+        if (fieldAnnotationMap == null) {
+            return Optional.empty();
+        }
+
+        Object[] keys = fieldAnnotationMap.getKeys();
+
+        for (Object key : keys) {
+            if (key instanceof BString bStringKey) {
+                String keyValue = bStringKey.getValue();
+                if (keyValue.endsWith(SENSITIVE_SUFFIX) && keyValue.startsWith(LOG_ANNOTATION_PREFIX)) {
```

Comment:
I do not completely understand this logic. Why are we checking the keyValue starts with `LOG_ANNOTATION_PREFIX`? How does this work?
Are we considering the cases where an alias is used for imports?